js/ui/grid_core/ui.grid_core.filter_row.js

```diff
@@ -670,17 +720,23 @@ const DataControllerFilterRowExtender = {
 
         const filters = [this.callBase()];
         const columns = this._columnsController.getVisibleColumns(null, true);
+        const filterRowController = this.getController('applyFilter');
 
         each(columns, function() {
-
-            if(this.allowFiltering && this.calculateFilterExpression && isDefined(this.filterValue)) {
+            const shouldSkip = filterRowController._currentColumn?.index === this.index;
```

Comment:
It is strange that _currentColumn is assigned to applyFilter controller that does not use this value.
Add public methods to applyFilter controller to get and set _currentColumn and use this public methods. Or _currentFilterColumn can be defined in dataController and can be the same for filterRow and headerFilter. We can add helper method getCombinedFilterWithoutColumn(column) to dataController to avoid code duplication for setting and clearing _currentColumn in several places.